Add config gem so we can use Settings.yml etc.

tingulfsen opened the pull request:
Fixes #21

jmartin-sul commented on config/initializers/config.rb:
```diff
@@ -0,0 +1,36 @@
+
+  # Parse numeric values as integers instead of strings.
+  #
+  # config.env_parse_values = true
```
on the one hand, this feels like it runs afoul of what we talked about this morning where we don't include commented out boilerplate.  on the other hand, i'd have been inclined to leave it also.  i don't think i have an opinion?  what do others think?

ndushay replied:
given that we actually do use configs a lot and I have never noticed all this boilerplate or mucked about with these settings, I would be comfortable removing all this boilerplate.  But I don't have strong feels and we could leave it, too.

ndushay commented:
In the end, I don't think we check in `development.yml` either after we have shared_configs, but we do need an example  .yml and README instructions on how to turn example.yml (or test.yml) into development.yml for local development.

tingulfsen commented:
I think I probably addressed the concerns mentioned above, rebased on master. I left development.yml and test.yml in there for the moment. We can remove those too if you like, but maybe it's easier to leave them since we know we will need them but don't have any offending data yet.

ndushay commented:
👍  to merge.  @jmartin-sul  I'll leave it to you ...
